### PR TITLE
Update to 71.0.3578.80

### PR DESCRIPTION
**Pull request by xsmile**
- removed `gn-bootstrap-remove-gn-gen.patch`, `--skip_generate_buildfiles` can now be used to skip this step
- ~~removed `add-flag-to-show-avatar-button.patch`, the new UI places the avatar button in line with the settings button instead of the title bar and switching to the old UI is not possible anymore~~

### CODE REVIEW

**Comment by ian-moone** [comment]
Thank you. It builds fine for me.

**Comment by qvint** [comment]
> * removed `add-flag-to-show-avatar-button.patch`, the new UI places the avatar button in line with the settings button instead of the title bar and switching to the old UI is not possible anymore

Does the patch stop working in version 71? Did you check that? It works for both UIs in version 70.

**Comment by perfect7gentleman** [comment]
```
 ~ $ chromium

(chromium-browser:17169): Gtk-WARNING **: 18:08:42.659: Theme parsing error: gtk.css:68:35: The style property GtkButton:child-displacement-x is deprecated and shouldn't be used anymore. It will be removed in a future version

(chromium-browser:17169): Gtk-WARNING **: 18:08:42.659: Theme parsing error: gtk.css:69:35: The style property GtkButton:child-displacement-y is deprecated and shouldn't be used anymore. It will be removed in a future version

(chromium-browser:17169): Gtk-WARNING **: 18:08:42.659: Theme parsing error: gtk.css:73:46: The style property GtkScrolledWindow:scrollbars-within-bevel is deprecated and shouldn't be used anymore. It will be removed in a future version
Aborted
```

**Comment by ian-moone** [comment]
@perfect7gentleman Those warnings comes from the theme engine, I think, and shouldn't cause an abort. On another issue tracker, I saw that you are using Clang as the default toolchain, yes? If so, you should double-check if you properly built chromium's dependencies, specially C++ ones.

**Comment by perfect7gentleman** [comment]
@ian-moone , Clang is default for chromium deps such as nodejs and others. New dep introduced in 71, jsoncpp, is also built with Clang.

**Comment by xsmile** [comment]
> Does the patch stop working in version 71? Did you check that? It works for both UIs in version 70.

I checked it briefly and it requires updating as code referencing the old UI was removed. If needed, I can update and include it again.

**Comment by ian-moone** [comment]
Have you successfully built chromium against libcxx/libxxabi in the past? Because I don't. I had to build `dev-libs/re2`, `dev-libs/jsoncpp` and of course, chromium itself with:

```
CXXFLAGS="${CFLAGS} -stdlib=libstdc++"
LDFLAGS="-Wl,-O2 -Wl,--as-needed -fuse-ld=lld -Wl,-lgcc_s"
```

**Comment by perfect7gentleman** [comment]
> ave you successfully built chromium against libcxx/libxxabi in the past?
yes, and lastest ungoogled-chromium-70 too.
> I had to build `dev-libs/re2`, `dev-libs/jsoncpp`
they are built with Clang

**Comment by ian-moone** [comment]
I'm not sure if chromium truly supports system libcxx on linux yet. At least looking at the `BUILD.gn`, libc++ are only related to Mac, like this one:

```
  if (is_mac) {
    # The system libc++ on Mac doesn't have aligned allocation in C++17.
    defines += [ "_LIBCPP_HAS_NO_ALIGNED_ALLOCATION" ]
    cflags_cc += [ "-stdlib=libc++" ]
    ldflags += [ "-stdlib=libc++" ]
  }
```

The other references are when `use_custom_libcxx == true`, which uses the bundled toolchain. So I suggest that you should stick to `libstdc++` for now.

**Comment by perfect7gentleman** [comment]
I don't see any big differences between ebuilds for 70 and 71. Only jsonccp as new dep.

**Comment by ian-moone** [comment]
Yeah, but chromium's internals could've change a lot between then.

**Comment by qvint** [comment]
@xsmile I can't rebuild chromium right now due to build errors. Could you please apply [the fixed patch](https://gist.github.com/qvint/18d53bd9af5b90a1598f4ab29346942f) and confirm that everything is OK?

On Fedora 29 I get the following errors (`linux_portable`):

```
$ (cd build/src && ./tools/gn/bootstrap/bootstrap.py -o out/Default/gn)
ninja: Entering directory `<...>/build/src/out/Release/gn_build'
[170/170] LINK gn
ERROR at //build/config/sysroot.gni:57:5: Assertion failed.
    assert(
    ^-----
Missing sysroot (//build/linux/debian_sid_amd64-sysroot). To fix, run: build/linux/sysroot_scripts/install-sysroot.py --arch=amd64
See //build/config/sysroot.gni:58:9: 
        exec_script("//build/dir_exists.py",
        ^-----------------------------------
This is where it was set.
See //base/BUILD.gn:29:1: whence it was imported.
import("//build/config/sysroot.gni")
^----------------------------------
See //BUILD.gn:66:5: which caused the file to be included.
    "//base:base_perftests",
    ^----------------------
Traceback (most recent call last):
  File "./tools/gn/bootstrap/bootstrap.py", line 101, in <module>
    sys.exit(main(sys.argv[1:]))
  File "./tools/gn/bootstrap/bootstrap.py", line 96, in main
    '--args=%s' % gn_gen_args, "--root=" + SRC_ROOT
  File "/usr/lib64/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['out/Default/gn', 'gen', '<...>/build/src/out/Release', '--args= is_debug=false', '--root=<...>/build/src']' returned non-zero exit status 1
```

**Comment by xsmile** [comment]
> @xsmile Please update the patch, [it is fixed now](https://gist.github.com/qvint/18d53bd9af5b90a1598f4ab29346942f).

Thank you, I included it again.

**Comment by Eloston** [comment]
Whoops, I only wanted 8c49d061de6fe764f8fc3d08547261114c4b41e4 to be approved again, not the entire PR...

File: patches/debian_buster/fixes/ambiguous-overloads.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
+description: gcc6 cannot automatically resolve these overloads, tell it what to do
```
[the remark this answers is not in the document]
Comment (xsmile) [reply patches/debian_buster/fixes/ambiguous-overloads.patch]: Fixed.

File: patches/debian_buster/system/fontconfig.patch [hunk 2]
```diff
@@ -1,16 +1,19 @@
-# CHANGES MADE for ungoogled-chromium: The import of
-# //buildtools/deps_revision.gni
-# in file build/config/posix/BUILD.gn
-# was reverted since it
-# - breaks linux_portable.
-# - has no clear side-effects (the include only defines constants).
-# - has nothing to do with fontconfig.
```
[the remark this answers is not in the document]
Comment (xsmile) [reply patches/debian_buster/system/fontconfig.patch]: Right, fixed.
Thanks for the instructions, I'll take a look next time.

File: patches/debian_buster/system/gtk2.patch [hunk 3]
```diff
@@ -0,0 +1,1655 @@
+description: restore support for building against gtk2
```
[the remark this answers is not in the document]
Comment (xsmile) [reply patches/debian_buster/system/gtk2.patch]: This is optional, I removed the patch.

File: patches/debian_buster/warnings/explicit-constructor.patch [hunk 4]
```diff
@@ -3,12 +3,11 @@ author: Michael Gilbert <mgilbert@debian.org>
 
 --- a/device/fido/hid/fido_hid_device.cc
 +++ b/device/fido/hid/fido_hid_device.cc
-@@ -40,7 +40,7 @@ void FidoHidDevice::Cancel() {
+@@ -40,7 +40,6 @@ void FidoHidDevice::Cancel() {
      return;
  
    // Delete any remaining pending requests on this Channel ID.
 -  pending_transactions_ = {};
-+  pending_transactions_ = base::queue<std::pair<std::vector<uint8_t>, DeviceCallback>>();
```
[the remark this answers is not in the document]
Comment (xsmile) [reply patches/debian_buster/warnings/explicit-constructor.patch]: Fixed.

**Pull request merged**